jgrapht-io/src/main/java/org/jgrapht/io/AbstractBaseImporter.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * (C) Copyright 2017-2017, by Dimitrios Michail and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.io;
+
+import java.util.Objects;
+
+import org.jgrapht.Graph;
+
+/**
+ * Base implementation for a graph importer.
+ *
+ * @param <V> the graph vertex type
+ * @param <E> the graph edge type
+ *
+ * @author Dimitrios Michail
+ * @since March 2017
+ */
+abstract class AbstractBaseImporter<V, E>
+{
+    /**
+     * Constructs new vertices
+     */
+    protected VertexProvider<V> vertexProvider;
+
+    /**
+     * Constructs new edges
+     */
+    protected EdgeProvider<V, E> edgeProvider;
+
+    /**
+     * Updates already constructed vertices
+     */
+    protected ComponentUpdater<V> vertexUpdater;
+
+    /**
+     * Updates graph properties
+     */
+    protected ComponentUpdater<Graph<V, E>> graphUpdater;
+
+    /**
+     * Constructor
+     *
+     * @param vertexProvider the vertex provider. Must not be null.
+     * @param edgeProvider the edge provider. Must not be null.
+     */
+    public AbstractBaseImporter(VertexProvider<V> vertexProvider, EdgeProvider<V, E> edgeProvider)
+    {
+        this(vertexProvider, edgeProvider, (c, a) -> {
+        }, (c, a) -> {
+        });
+        this.vertexProvider =
```

Comment:
vertexProvider and edgeProvider have already been initialized by the preceding call to this(...)